lib/cylc/external_trigger.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2015 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import time
+import logging
+import Queue
+import Pyro.core
+import Pyro.errors
+import cylc.flags
+import cylc.cylc_pyro_client
+from cylc.task_id import TaskID
+from cylc.passphrase import passphrase
+
+
+PYRO_TARGET_NAME = 'external_event_broker'
+
+
+class PyroClient(object):
+    """Network client for sending external triggers to a suite."""
+
+    # TODO - configurable network retry parameters?
+    MAX_TRIES = 5
+    PYRO_TIMEOUT = 10
+    RETRY_SECONDS = 10
+
+    def __init__(
+            self, suite, event_msg, host='localhost', owner=None,
+            port=None, event_id=None):
+
+        self.event_msg = event_msg
+        self.event_id = event_id
+        self.suite = suite
+        self.host = host
+        self.owner = owner
+        self.port = port
+
+        self.pphrase = passphrase(
+                self.suite, self.owner, self.host).get(None, None)
+
+        self.pyro_proxy = cylc.cylc_pyro_client.client(
+                self.suite, self.pphrase,
+                self.owner, self.host, self.__class__.PYRO_TIMEOUT,
+                self.port ).get_proxy(PYRO_TARGET_NAME)
+
+    def send(self):
+        sent = False
+        itry = 0
+        while True:
+            itry += 1
+            try:
+                # Get a proxy for the remote object and send the message.
+                self.pyro_proxy.register(self.event_msg, self.event_id)
+            except Pyro.errors.NamingError as exc:
+                print >> sys.stderr, exc
+                print "Send message: try %s of %s failed: %s" % (
```

Comment:
This message could be a class level variable as it's used several times e.g. `ERROR_FAILED_SEND="Send message: try %s of %s failed: %s"`.